File: Backend/controllers/predictionInputController.js
const PredictionInput = require('../models/predictionInputModel');

const PredictionInputController = {
  getAll: async (req, res) => {
    try {
      const data = await PredictionInput.findAll(req.user.role === 'admin' ? null : req.user.id);
      res.json({ success: true, data });
    } catch (error) {
      res.status(500).json({ success: false, message: 'Server error' });
    }
  },
  getStats: async (req, res) => {
    try {
      const stats = await PredictionInput.getStats(req.user.id);
      res.json({ success: true, data: stats });
    } catch (error) {
      res.status(500).json({ success: false, message: 'Server error' });
    }
  }
};

module.exports = PredictionInputController;

File: backend_python/controllers/prediction_input_controller.py
from flask import g, request

from models.prediction_input_model import PredictionInput
from utils.helpers import fail, ok


class PredictionInputController:
    @staticmethod
    def get_all():
        user_id = None if g.user.get("role") == "admin" else g.user["id"]
        return ok({"success": True, "data": PredictionInput.find_all(user_id)})

    @staticmethod
    def get_stats():
        return ok({"success": True, "data": PredictionInput.get_stats(g.user["id"])})
